wasm/src/main/scala/wasm4s/WasmContext.scala

```diff
@@ -818,51 +800,60 @@ object WasmContext {
 
     def isInterface = kind == ClassKind.Interface
 
-    def methods: List[WasmFunctionInfo] = _methods
-
-    def maybeAddAbstractMethod(methodName: IRNames.MethodName, ctx: WasmContext): Unit = {
-      if (!methods.exists(_.name.simpleName == methodName.nameString)) {
-        val wasmName = WasmFunctionName(IRTrees.MemberNamespace.Public, name, methodName)
-        val argTypes = methodName.paramTypeRefs.map(ctx.inferTypeFromTypeRef(_))
-        val resultType = ctx.inferTypeFromTypeRef(methodName.resultTypeRef)
-        _methods = _methods :+ WasmFunctionInfo(
-          wasmName,
-          argTypes,
-          resultType,
-          isAbstract = true,
-          isReflectiveProxy = methodName.isReflectiveProxy
-        )
+    def registerDynamicCall(methodName: IRNames.MethodName): Unit =
+      methodsCalledDynamically += methodName
+
+    def buildMethodTable(): Unit = {
+      if (_tableEntries != null)
+        throw new IllegalStateException(s"Duplicate call to buildMethodTable() for $name")
+
+      kind match {
+        case ClassKind.Class | ClassKind.ModuleClass | ClassKind.HijackedClass =>
+          val superClassInfo = superClass.map(ctx.getClassInfo(_))
+          val superTableEntries =
+            superClassInfo.fold[List[IRNames.MethodName]](Nil)(_.tableEntries)
```

Comment:
Yes, this is not the first thing that requires such a topological order. It is enforced by sorting classes by their `.ancestors.size` at
https://github.com/tanishiking/scala-wasm/blob/1be5a2b46be6f99652cb1b6afc634797a20a685d/wasm/src/main/scala/WebAssemblyLinkerBackend.scala#L81-L88
We also need this order for building the list of fields of a class, for example. Indirectly we also need it to generate the `type`s of the `struct`s for the classes, since they declare a subtyping relationship at the Wasm level and therefore must appear in inheritance order.

So yes, it's pretty pervasive and we can/have to rely on it.